Formula/couchdb.rb

```diff
@@ -19,14 +19,31 @@ class Couchdb < Formula
   depends_on "erlang@22" => :build
   depends_on "libtool" => :build
   depends_on "pkg-config" => :build
+  depends_on "gcc" # for c++20
   depends_on "icu4c"
   depends_on "openssl@1.1"
   depends_on "spidermonkey"
 
   conflicts_with "ejabberd", because: "both install `jiffy` lib"
 
+  patch :DATA
+
+  # feat(couchjs): add support for SpiderMonkey 86
+  patch do
+    url "https://github.com/apache/couchdb/commit/a411fe220f94be73b9364fc0e923b8401d4e73ff.patch?full_index=1"
+    sha256 "1bc52ae8f602701a27262a4174006306fa1d357d41944d82b3061bf7b541639b"
+  end
+
+  # Add support for Spidermonkey 78
+  patch do
+    url "https://github.com/apache/couchdb/commit/37b5eed5024b9a57a134260794ea80032a1149f8.patch?full_index=1"
+    sha256 "caf4a73d6e0c7edbe508be94d9ca47da3cc65bf3900cb22151c2094bba7ba9cb"
+  end
+
   def install
-    system "./configure"
+    spidermonkey = Formula["spidermonkey"]
+    ENV["HOMEBREW_CC"] = Formula["gcc"].bin/"gcc-#{Formula["gcc"].version.major}"
```

Comment:
This isn't the right way to force use of `gcc` to compile. Use `fails_with` for that. We can probably also restrict compiling with `gcc` to Mojave, no? We may also wish to consider using `llvm`'s `clang` to avoid adding a new runtime dependency.